books.c

```diff
@@ -277,5 +333,62 @@ void freeBookList(Book* head)
     }
 }
 
+int deleteBook(char* input_ISBN)     // Delete source book
+{
+    Book* previous = NULL;
+    Book* current = library;
+
+    if (!current) {
+        printf("Library is empty.\n");
+        return 0;
+    }
+
+    while (current) {
+        if (isMatch(current->ISBN, input_ISBN)) {
+            if (!previous) { // if match @ head
+                library = current->next;
+            } else {
+                previous->next = current->next;
+            }
+            free(current);
+            return 1; // Book found and deleted, return because there should only 1 book match
```

Comment:
As discussed in a live class, we are not going to delete record permanently from our system, we will prefer soft delete (status: inactive/deleted)